src/npcmove.cpp

```diff
@@ -2253,16 +2256,20 @@ void npc::go_to_destination()
         // sx and sy are now equal to the direction we need to move in
         int x = posx() + 8 * sx, y = posy() + 8 * sy;
         // x and y are now equal to a local square that's close by
-        for (int i = 0; i < 8; i++) {
-            for (int dx = 0 - i; dx <= i; dx++) {
-                for (int dy = 0 - i; dy <= i; dy++) {
-                    if ((g->m.move_cost(x + dx, y + dy) > 0 ||
+        tripoint dest;
+        dest.z = posz();
+        int &dx = dest.x;
+        int &dy = dest.y;
+        for( int i = 0; i < 8; i++ ) {
+            for( dx = x - i; dx <= x + i; dx++ ) {
+                for( dy = y - i; dy <= y + i; dy++ ) {
+                    if( ( g->m.move_cost( dest ) > 0 ||
                          //Needs 20% chance of bashing success to be considered for pathing
-                         g->m.bash_rating(str_cur + weapon.type->melee_dam, x, y) >= 2 ||
```

Comment:
Nice, you fixed the bug here: should never have been `x, y` but `dx+x, dy+y` :+1: